source/request_response.c

```diff
@@ -133,15 +133,17 @@ void aws_http_request_destroy(struct aws_http_request *request) {
     aws_string_destroy(request->method);
     aws_string_destroy(request->path);
 
-    const size_t length = aws_array_list_length(&request->headers);
-    struct aws_http_header_impl *header_impl = NULL;
-    for (size_t i = 0; i < length; ++i) {
-        aws_array_list_get_at_ptr(&request->headers, (void **)&header_impl, i);
-        AWS_ASSERT(header_impl);
-        s_header_impl_clean_up(header_impl);
-    }
+    if (aws_array_list_is_valid(&request->headers)) {
```

Comment:
This makes me nervous. If we need to check array lists for validity before cleaning them up (which I know I don't do all over the place) due to construction-OOM failures then if possible I'd prefer that array_list init is all-or-nothing like C++ constructors (if possible).

Reply:
+1

Reply:
This code-change was handled in https://github.com/awslabs/aws-c-http/pull/94